# Iris Flower Classification 

## Importing relevant libraries 

In [3]:
import sklearn
import numpy as np
import pandas as pd

## Loading the dataset

In [5]:
data_init = sklearn.datasets.load_iris()

In [6]:
data_init

{'data': array([[5.1, 3.5, 1.4, 0.2],
        [4.9, 3. , 1.4, 0.2],
        [4.7, 3.2, 1.3, 0.2],
        [4.6, 3.1, 1.5, 0.2],
        [5. , 3.6, 1.4, 0.2],
        [5.4, 3.9, 1.7, 0.4],
        [4.6, 3.4, 1.4, 0.3],
        [5. , 3.4, 1.5, 0.2],
        [4.4, 2.9, 1.4, 0.2],
        [4.9, 3.1, 1.5, 0.1],
        [5.4, 3.7, 1.5, 0.2],
        [4.8, 3.4, 1.6, 0.2],
        [4.8, 3. , 1.4, 0.1],
        [4.3, 3. , 1.1, 0.1],
        [5.8, 4. , 1.2, 0.2],
        [5.7, 4.4, 1.5, 0.4],
        [5.4, 3.9, 1.3, 0.4],
        [5.1, 3.5, 1.4, 0.3],
        [5.7, 3.8, 1.7, 0.3],
        [5.1, 3.8, 1.5, 0.3],
        [5.4, 3.4, 1.7, 0.2],
        [5.1, 3.7, 1.5, 0.4],
        [4.6, 3.6, 1. , 0.2],
        [5.1, 3.3, 1.7, 0.5],
        [4.8, 3.4, 1.9, 0.2],
        [5. , 3. , 1.6, 0.2],
        [5. , 3.4, 1.6, 0.4],
        [5.2, 3.5, 1.5, 0.2],
        [5.2, 3.4, 1.4, 0.2],
        [4.7, 3.2, 1.6, 0.2],
        [4.8, 3.1, 1.6, 0.2],
        [5.4, 3.4, 1.5, 0.4],
        [5.2, 4.1, 1.5, 0.1],
  

Information on the data shows that it has 4 attributes:
- Sepal Length
- Sepal Width
- Petal Length
- Petal Width

In [8]:
data = np.array(data_init["data"])

## Exploratory Data Analysis 

In [11]:
# Checking the shape
data.shape

(150, 4)

The data has 150 items and 4 attributes. As described earlier, it has 50 in each i.e setosa, Versicolour, and Virginica.

In [13]:
# Checking data types
data.dtype

dtype('float64')

In [14]:
## using initial feature names
feature_names = data_init.feature_names
feature_names

## missing values
pd.DataFrame(data, columns=feature_names).isnull().sum()

sepal length (cm)    0
sepal width (cm)     0
petal length (cm)    0
petal width (cm)     0
dtype: int64

### Making a copy for manipulation 

In [16]:
data_arr = data.copy()

In [17]:
data_split = data_arr.reshape(3, 50, 4)

In [19]:
setosa = data_split[0]
versicolour = data_split[1]
virginica = data_split[2]

### Making pandas dataframes for setosa, versicolour and viriginica

In [21]:
setosa_df = pd.DataFrame(setosa, columns=feature_names)
versicolour_df = pd.DataFrame(versicolour, columns=feature_names)
virginica_df =pd.DataFrame(virginica, columns=feature_names)

In [22]:
setosa_df["target"] = "setosa"
versicolour_df["target"] = "versicolour"
virginica_df["target"] = "virginica"

df = pd.concat([setosa_df,versicolour_df,virginica_df])
df

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa
...,...,...,...,...,...
45,6.7,3.0,5.2,2.3,virginica
46,6.3,2.5,5.0,1.9,virginica
47,6.5,3.0,5.2,2.0,virginica
48,6.2,3.4,5.4,2.3,virginica


### Descriptive Statisitcs of Setosa

In [24]:
for x in feature_names:
    print("========================================")
    print(setosa_df[x].describe(include=all))

count    50.00000
mean      5.00600
std       0.35249
min       4.30000
25%       4.80000
50%       5.00000
75%       5.20000
max       5.80000
Name: sepal length (cm), dtype: float64
count    50.000000
mean      3.428000
std       0.379064
min       2.300000
25%       3.200000
50%       3.400000
75%       3.675000
max       4.400000
Name: sepal width (cm), dtype: float64
count    50.000000
mean      1.462000
std       0.173664
min       1.000000
25%       1.400000
50%       1.500000
75%       1.575000
max       1.900000
Name: petal length (cm), dtype: float64
count    50.000000
mean      0.246000
std       0.105386
min       0.100000
25%       0.200000
50%       0.200000
75%       0.300000
max       0.600000
Name: petal width (cm), dtype: float64


### Descriptive Statisitcs of Versicolour

In [26]:
for x in feature_names:
    print("========================================")
    print(versicolour_df[x].describe(include=all))

count    50.000000
mean      5.936000
std       0.516171
min       4.900000
25%       5.600000
50%       5.900000
75%       6.300000
max       7.000000
Name: sepal length (cm), dtype: float64
count    50.000000
mean      2.770000
std       0.313798
min       2.000000
25%       2.525000
50%       2.800000
75%       3.000000
max       3.400000
Name: sepal width (cm), dtype: float64
count    50.000000
mean      4.260000
std       0.469911
min       3.000000
25%       4.000000
50%       4.350000
75%       4.600000
max       5.100000
Name: petal length (cm), dtype: float64
count    50.000000
mean      1.326000
std       0.197753
min       1.000000
25%       1.200000
50%       1.300000
75%       1.500000
max       1.800000
Name: petal width (cm), dtype: float64


### Descriptive Statisitcs of Virginica

In [28]:
for x in feature_names:
    print("========================================")
    print(virginica_df[x].describe(include=all))

count    50.00000
mean      6.58800
std       0.63588
min       4.90000
25%       6.22500
50%       6.50000
75%       6.90000
max       7.90000
Name: sepal length (cm), dtype: float64
count    50.000000
mean      2.974000
std       0.322497
min       2.200000
25%       2.800000
50%       3.000000
75%       3.175000
max       3.800000
Name: sepal width (cm), dtype: float64
count    50.000000
mean      5.552000
std       0.551895
min       4.500000
25%       5.100000
50%       5.550000
75%       5.875000
max       6.900000
Name: petal length (cm), dtype: float64
count    50.00000
mean      2.02600
std       0.27465
min       1.40000
25%       1.80000
50%       2.00000
75%       2.30000
max       2.50000
Name: petal width (cm), dtype: float64


## Visualizations 